### PR TITLE
fix: support grafana 8.3.x

### PR DESCRIPTION
**Pull request by geryepes**
Upstream changes related to this PR
https://github.com/grafana/grafana/commit/7319efe0779d97c8dac5b0e6d9a3749b177c45a0

### CODE REVIEW

**Comment by geryepes** [comment]
Resolves #139

**Comment by yggdrasil-tynor** [comment]
@geryepes For me it looks good. However it would be better for a regular contributor to make the review, as i haven't worked on grafana plugins. 

Nice work by the way :)

**Comment by vicioussn** [comment]
Anyone? Please merge it.

**Comment by Gauravshah** [comment]
I can look into after 7 Days. On PTO

**Comment by yggdrasil-tynor** [comment]
@Gauravshah would you be happy to review it? :-)

File: dist/datasource.js [hunk 1]
```diff
@@ -204,7 +211,13 @@ function (angular, _, dateMath, moment) {
...
             }
             else{
-                return ds.query(options)
+                var ds_res = ds.query(options);
+                if(ds_res.then){
+                    return ds_res;
+                }
+                else{
+                    return ds_res.toPromise();
+                }
```
Comment (Gauravshah) [inline dist/datasource.js:220]: unrelated changes, there is another PR here https://github.com/GoshPosh/grafana-meta-queries/pull/132/files

**Review by Gauravshah — CHANGES_REQUESTED** [review]
Thank you for putting in PR and also linking the upstream grafana change

File: dist/datasource.js (hunk 1 above)
Comment (Gauravshah) [reply dist/datasource.js:220]: lets keep this PR specific to this change, will look into reviewing other one as well

Comment (ryanpxyz) [reply dist/datasource.js:220]: Thanks for taking a look. 

Quick question: Can you provide any kind of prediction as to when this change might become generally available? We have a broken 'datasource' currently and I really don't want to rollback by Grafana version. My users are also looking for alternatives to MetaQuery ...

Comment (Gauravshah) [reply dist/datasource.js:220]: have asked the original author to simplify PR https://github.com/GoshPosh/grafana-meta-queries/pull/128, if I do not hear back in a day, will create a branch from author's commit and merge it

Comment (Gauravshah) [reply dist/datasource.js:220]: @ryanpxyz https://github.com/GoshPosh/grafana-meta-queries/pull/128 should solve here right ?

Comment (geryepes) [reply dist/datasource.js:220]: > lets keep this PR specific to this change, will look into reviewing other one as well

Oh I didn't see that. I will work on it in a few hours

Comment (ryanpxyz) [reply dist/datasource.js:220]: @Gauravshah For you information: We are currently running - 

Grafana v8.3.4
Grafana Operator v4.1.1

We get this error message from 'MetaQueries': "'Datasource [object Object] was not found'"

I am hoping that PR #141 fixes this.

**Comment by ryanpxyz** [comment]
Good morning, 

Is there a status update? Are there plans to release a 0.0.6?

File: dist/datasource.js (hunk 1 above)
Comment (geryepes) [reply dist/datasource.js]: @Gauravshah changes done

**Review by Gauravshah — APPROVED** [review]
@ShilpaSivanesan can we merge this ?

**Comment by ShilpaSivanesan** [comment]
> @ShilpaSivanesan can we merge this ?

Sure, will validate and merge

**Comment by yggdrasil-tynor** [comment]
Can this be merged? :)

**Comment by yggdrasil-tynor** [comment]
@Gauravshah / @ShilpaSivanesan  sorry to bump you guys. But can this please be merged, so we can use this great plugin with grafana 8.4.0 👍

**Comment by ryanpxyz** [comment]
Good morning, 

Is there a status update? Are there plans to release a 0.0.6?

**Comment by ShilpaSivanesan** [comment]
Thanks for your contribution @geryepes . Really appreciate it.
